src/components/Nav/Nav.js

```diff
@@ -1,7 +1,360 @@
-import React from 'react';
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { useHistory } from 'react-router-dom';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSearch } from '@fortawesome/free-solid-svg-icons';
+import { faShoppingCart } from '@fortawesome/free-solid-svg-icons';
+import { faUserAlt } from '@fortawesome/free-solid-svg-icons';
 
 function Nav() {
-  return <div></div>;
+  const entry = Object.entries(DATA);
+  const [selectedKey, setSelectedKey] = useState();
+  const [modal, setModal] = useState(false);
+  const history = useHistory();
+  const goToSignUp = () => {
+    history.push('/sign');
+  };
+
+  return (
+    <>
+      <Ad>Ader Error x CASETiFANY 출시 구매하기</Ad>
+      <NavContainter>
+        <a href="/">
+          <Logo src="/images/CASTIFANY.png" />
+        </a>
+        <NavList>
+          {entry.map(el => (
+            <NavListContent onMouseOver={() => setSelectedKey(el[0])}>
+              {el[0]}
+            </NavListContent>
+          ))}
+        </NavList>
+        <NavIcon>
+          <NavIconContent>
+            <FontAwesomeIcon icon={faSearch} />
+          </NavIconContent>
+          <NavIconContent>
+            <FontAwesomeIcon icon={faShoppingCart} />
+          </NavIconContent>
+          <UserBox>
+            <NavIconContent>
+              <FontAwesomeIcon
+                icon={faUserAlt}
+                // !token? {goToSignUp}:
+                onClick={() => {
+                  setModal(!modal);
+                }}
+              />
+            </NavIconContent>
+            {modal === true ? (
+              <UserInfo>
+                <UserInfoList>내 갤러리</UserInfoList>
+                <UserInfoList>설정</UserInfoList>
+                <UserInfoList>주문 현황</UserInfoList>
+                <UserInfoList>지원</UserInfoList>
+                <UserInfoList>로그아웃</UserInfoList>
+              </UserInfo>
+            ) : null}
+          </UserBox>
+        </NavIcon>
+      </NavContainter>
+      <NavModalContainer>
+        <NavModal onMouseLeave={() => setSelectedKey()}>
+          <NavModalContent>
+            <NavModalWrap>
+              {selectedKey &&
+                DATA[selectedKey]['productList'].map(el => {
```

Comment:
```suggestion
                DATA[selectedKey].productList.map(el => {
```
두번째 부분은 bracket notation으로 적용할 필요 없어보입니다!